### DISCUSSION TITLE
Valgrind - assertion failed: Calling init too late

### DISCUSSION THREAD

**Discussion by stevengogogo** [stevengogogo/DSA_EmailSearcher#21] [Q&A] [answered]
I got this error by analyzing `main`, within template folder, with valgrind:

```
assertion failed: Calling init too late
```

Here is the screenshot of my workspace:
<img width="2560" alt="Screen Shot 2021-05-20 at 5 08 01 PM" src="https://user-images.githubusercontent.com/29009898/118952081-3758e780-b98e-11eb-95f8-e57b7bcddc05.png">

**Comment by stevengogogo** [comment]
Email
------
> 您好,
> 我在測試 template 產生的 main file 時, 發現 valgrind 會有以下訊息
> assertion failed: Calling init too late

> 下圖為工作視窗截圖, 請問這是正常的嗎?

**Comment by jcgs2503** [comment, marked as answer]
Not quite sure, but I think it's related to the 15ms or 30ms limitation

**Reply by stevengogogo** [reply]
That makes sense! We can

1. Ignore / use `make LEAK` to test our project (**Preferred**)
2. Remove the time limit assertion for testing (I'll open a branch to test it)

**Reply by stevengogogo** [reply]
You are right:

[`api.c`]
https://github.com/stevengogogo/DSA_EmailSearcher/blob/4c1cf0095f7b6f028e76be9f41c775f5452db618/src/api.c#L42

[Action]
https://github.com/stevengogogo/DSA_EmailSearcher/runs/2628410914
> Somehow it doen't print out the e`Calling init too late"`
> But I'm sure it happens on terminal

#22

**Comment by stevengogogo** [comment]
#22  is solved! Thanks @jcgs2503
